File: 6_kyu/Shortest_steps_to_a_number.go
package kata

func ShortestStepsToNum(n int) int {
  var counter int = 0
  for n > 1 {
    if n & 1 > 0 {
      counter += 1
      n -= 1
    }
    counter += 1
    n /= 2
  }
  return counter
}

File: 6_kyu/Shortest_steps_to_a_number.rs
fn shortest_steps_to_num(n: u16) -> u16 {
    let mut counter = 0;
    let mut remain = n;
    while remain > 1 {
        if remain & 1 > 0 {
            counter += 1;
        }
        counter += 1;
        remain /= 2;
    }
    counter
}
